### DISCUSSION TITLE
Command Tasmota plug from ESP32 in AP mode

### DISCUSSION THREAD

**Discussion by risoul** [arendst/Tasmota#19275] [Q&A] [answered]
Hi,

Here is my configuration:
- ESP32 with Access Point ON
- Tasmota plug (Athom) connected on that AP
- ESP32 list all the connected devices and I can see the Athom Plug (192.168.4.2)

My goal is to pilot Athom plug from ESP32 with only AP (no router or internet connection).

Sending the command by:

```
    String url = "http://192.168.4.2/cm?cmnd=Power%20On";
    HTTPClient http;
    http.begin(url);
    int httpCode = http.GET();
    http.end();
```
does nothing, same for Off.

Connecting my laptop to the AP and navigate to http://192.168.4.2/cm?cmnd=Power%20On does nothing also.

What am I doing wrong?

Thanks for your help

**Comment by sfromis** [comment]
Whenever asking about "does nothing" or similar, you should instead "look better" at what happens, generally by watching the console after having activated log level 3 for details. If the log does not get anything, the answer is outside of Tasmota.

**Reply by risoul** [reply]
Sorry, by "does nothing" I mean, the state of the relay is not changing.

You are right, I have this error in my console : [2001089][E][WiFiClient.cpp:268] connect(): socket error on fd 48, errno: 104, "Connection reset by peer"

**Reply by sfromis** [reply]
I meant in the Tasmota console, not your non-Tasmota device. It is irrelevant to say "does nothing" without having looked for details of what it is doing, other than what you wanted. And with log level 3.

**Reply by risoul** [reply]
Thanks, but how can I access the console? since the Athom plug is connected to ESP32 AP and no longer appears on my devices, I cannot connect to it ? 
I only see it on 192.168.4.2 by scanning the network when connected at the same AP, I can ping it with succes, but nothing else.

**Reply by sfromis** [reply]
Since you tried doing the request from your browser, that would mean that you have a browser with http access.

**Reply by risoul** [reply]
I just tried to put the url http://192.168.4.2/cm?cmnd=Power%20On in my browser, but I have no interface.

**Reply by risoul** [reply]
[TLDR]
Can an Athom socket with the factory version of Tasmota installed on it (not reflashed) should work when connected only to a hotspot from a smartphone with 4G enable?
-----------------

I'll try to be more clear.

First I tried to connect the Athom socket to my ESP32 configured as an AP only.

Everything is going well, it connects, its wifi disappears from my smartphone.
By connecting to the ESP32 from my smartphone and scanning the network I can see the Athom socket at the address 192.168.4.2. When I ping this address, I have a response and zero packets lost.
But impossible to send a command to the outlet.
I tried to detect the socket with the Tasmotrol Android application. It can't find any equipment.

I wondered if the problem was not from the AP on ESP32.

So then I connected the Athom socket directly to the hotspot of my smartphone.
It connects well, I see it by scanning the network, but impossible to detect it with Tasmotrol again...

I think the problem is not with my initial code, but rather the refusal of the plug to work connected to a hotspot.
However, I have 2 TPlink sockets currently connected on it which work perfectly.

Can you confirm that Athom sockets are supposed to work when connected to a smartphone hotspot?
https://templates.blakadder.com/athom_PG01EU16A.html
It comes with Tasmota preinstalled, but may be it is a modified version that does not work with hostspots?

**Reply by risoul** [reply]
Sorry, i realize I forgot to give you the error message from ESP32 console when send a command to the Athom plug :+1: 

socket error on fd 48, errno: 104, "Connection reset by peer"

**Comment by arendst** [comment]
As your tasmota AP provides address 192.168.4.2 it won't be able to support commands; it's still in the onboarding phase. You cannot perform commands in this phase.

**Reply by risoul** [reply]
192.168.4.2 was the local IP of the Athom plug when connected directly to the ESP32 AP only witch is 192.168.4.1.

If I connect it to my phone AP (I do not have a router, only use a phone hostpot), I can also see it on the network at 192.168.43.136 (with the name Outlet_0454A0).  I can ping it with succes, but Tasmotrol (Android) does not detect it, Tasmoadmin from my laptop can't find it either.

But writing this I have some news. I was scanning port on 192.168.43.136 using Network Scanner (Android) and suddenly i heard the relay of the plug switching... Network scanner show only 5556 port opened (as Freeciv, Oracle Weblogic Server Node) and sometimes port 80 (HTTP).

Then going to 192.168.43.136 from my laptop give me another athom menu with reset, or upgrade from bin.
![Capture d’écran_2023-08-08_18-48-50](https://github.com/arendst/Tasmota/assets/10132802/e3bc7a0c-8f28-400c-ae99-6b251b8f2ab9)

I was unable to access this menu before.

So now I know that I can access the plug, I do not know why only after scanning ports...

But doing curl http://192.168.43.136/cm?cmnd=Power%20On from my laptop console does not switch and return "not found"

**Reply by risoul** [reply]
I can now confirm, after a reset and pairing, that I cannot access the admin interface on 192.168.43.136 until I have scanned the ports... Strange behaviour. When scanning ports, the led is blinking then I can acces the webui.

That simple "command a Tasmota plug from ESP32 with AP only" project is becoming a bit to hard for me...

**Reply by risoul** [reply]
Ok! finally i decided to upgrade with the firmware bin.gz here : https://github.com/athom-tech/athom-configs
Then I can connected to the webUI
The Athom athom_PG01EU16A was not in the device list, so I created it following this : https://templates.blakadder.com/athom_PG01EU16A.html for GPIO
Now I can switch the plug using curl from console.

Two little problems left:
- voltage, current and other measures stay to 0
- the led is not reflecting the state, stay on

Next step, test it with ESP32 sending commands.

Thanks for all that work you did, Tasmota is just amazing! I just do not understand why Athom sell the plugs with a modified version of Tasmota not working fine. Problem solved by upgrade to full version.

For my project, I plan to sell smartplugs with an ESP32 controler, can you advise me about brands and sellers with full Tasmota preinstalled, not just a modified version like Athom ?

**Reply by ferbulous** [reply]
There's shelly and switchbot plugs that can be flashed via ota.

**Comment by Jason2866** [comment, marked as answer]
You don't have Tasmota on your plug. You have Homekit firmware
